src/backend/vector/neon/edwards.rs

```diff
@@ -0,0 +1,547 @@
+
+        let mut tmp = self.0;
+
+        tmp = tmp.blend(tmp.diff_sum(), Lanes::AB);
+        // tmp = (Y1-X1 Y1+X1 Z1 T1) = (S0 S1 Z1 T1) with b < 1.6
+
+        // (tmp, other) bounded with b < (1.6, 1.0) < (2.5, 1.75).
+        tmp = &tmp * &other.0;
+        // tmp = (S0*S2' S1*S3' Z1*Z2' T1*T2') = (S8 S9 S10 S11)
+
+        tmp = tmp.shuffle(Shuffle::ABDC);
+        // tmp = (S8 S9 S11 S10)
+
+        tmp = tmp.diff_sum();
+        // tmp = (S9-S8 S9+S8 S10-S11 S10+S11) = (S12 S13 S14 S15)
+
+        let t0 = tmp.shuffle(Shuffle::ADDA);
+        // t0 = (S12 S15 S15 S12)
+        let t1 = tmp.shuffle(Shuffle::CBCB);
+        // t1 = (S14 S13 S14 S13)
+
+        // All coefficients of t0, t1 are bounded with b < 1.6.
+        // Return (S12*S14 S15*S13 S15*S14 S12*S13) = (X3 Y3 Z3 T3)
+        ExtendedPoint(&t0 * &t1)
+    }
+}
+
+impl<'a, 'b> Sub<&'b CachedPoint> for &'a ExtendedPoint {
+    type Output = ExtendedPoint;
+
+    /// Implement subtraction by negating the point and adding.
+    ///
+    /// Empirically, this seems about the same cost as a custom
+    /// subtraction impl (maybe because the benefit is cancelled by
+    /// increased code size?)
+    fn sub(self, other: &'b CachedPoint) -> ExtendedPoint {
+        self + &(-other)
+    }
+}
+
+impl<'a> From<&'a edwards::EdwardsPoint> for LookupTable<CachedPoint> {
+    fn from(point: &'a edwards::EdwardsPoint) -> Self {
+        let P = ExtendedPoint::from(*point);
+        let mut points = [CachedPoint::from(P); 8];
+        for i in 0..7 {
+            points[i + 1] = (&P + &points[i]).into();
+        }
+        LookupTable(points)
+    }
+}
+
+impl<'a> From<&'a edwards::EdwardsPoint> for NafLookupTable5<CachedPoint> {
+    fn from(point: &'a edwards::EdwardsPoint) -> Self {
+        let A = ExtendedPoint::from(*point);
+        let mut Ai = [CachedPoint::from(A); 8];
+        let A2 = A.double();
+        for i in 0..7 {
+            Ai[i + 1] = (&A2 + &Ai[i]).into();
+        }
+        // Now Ai = [A, 3A, 5A, 7A, 9A, 11A, 13A, 15A]
+        NafLookupTable5(Ai)
+    }
+}
+
+impl<'a> From<&'a edwards::EdwardsPoint> for NafLookupTable8<CachedPoint> {
+    fn from(point: &'a edwards::EdwardsPoint) -> Self {
+        let A = ExtendedPoint::from(*point);
+        let mut Ai = [CachedPoint::from(A); 64];
+        let A2 = A.double();
+        for i in 0..63 {
+            Ai[i + 1] = (&A2 + &Ai[i]).into();
+        }
+        // Now Ai = [A, 3A, 5A, 7A, 9A, 11A, 13A, 15A, ..., 127A]
+        NafLookupTable8(Ai)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    fn serial_add(P: edwards::EdwardsPoint, Q: edwards::EdwardsPoint) -> edwards::EdwardsPoint {
+        use backend::serial::u64::field::FieldElement51;
+
+        let (X1, Y1, Z1, T1) = (P.X, P.Y, P.Z, P.T);
+        let (X2, Y2, Z2, T2) = (Q.X, Q.Y, Q.Z, Q.T);
+
+        macro_rules! print_var {
+            ($x:ident) => {
+                println!("{} = {:?}", stringify!($x), $x.to_bytes());
```

Comment:
```suggestion
                println!("{} = {:?}", stringify!($x), $x.as_bytes());
```

These wer renamed to as_bytes()